src/chocolatey/infrastructure.app/nuget/NugetList.cs

```diff
@@ -43,11 +44,12 @@ public static IEnumerable<IPackage> GetPackages(ChocolateyConfiguration configur
                 results = results.Where(p => p.IsLatestVersion);
             }
 
-            return results.OrderBy(p => p.Id)
-                          .AsEnumerable()
-                          .Where(PackageExtensions.IsListed)
-                          .Where(p => configuration.Prerelease || p.IsReleaseVersion())
-                          .distinct_last(PackageEqualityComparer.Id, PackageComparer.Version).ToList();
+            // just trust the server order, don't sort because that's a blocking operation
+            // also don't worry about multiple versions, considering Is*LatestVersion only applies to one

```

Comment:
Well, distinct_last doesn't inherently block streaming: it returns the last element of one set as soon as it sees the first element of the next set -- but it requires sorting. Unless the sorting is done server side, you have to block to sort.